src/wallet/coinselection.cpp

```diff
@@ -452,12 +452,16 @@ void SelectionResult::AddInputs(const std::set<COutput>& inputs, bool subtract_f
 
 void SelectionResult::Merge(const SelectionResult& other)
 {
+    // Obtain the expected selected inputs count after the merge (for now, duplicates are not allowed)
```

Comment:
```suggestion
    // Store sum of combined input sets to check that no UTXO was used twice
```

Reply:
this relies on https://github.com/bitcoin/bitcoin/pull/26560#discussion_r1038307694 (where we are talking about whether should include the assertion commit here or in a follow-up PR).
Please, feel free to comment there.